Pointer Lock should compile on iOS

aprotyas opened the pull request:
#### 891c03ce502980be5aac97e1e09f2c088b255cb9
<pre>
Pointer Lock should compile on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=296501">https://bugs.webkit.org/show_bug.cgi?id=296501</a>
<a href="https://rdar.apple.com/156741590">rdar://156741590</a>

Reviewed by Richard Robinson and Aditya Keerthi.

This patch drops the build-time platform restriction, since there is
nothing in the Pointer Lock support code that prevents it from building
on most iOS family platforms.

Note that we still don&apos;t build Pointer Lock on tvOS/watchOS because of
missing platform support.

* Source/WTF/wtf/PlatformEnableCocoa.h:

Canonical link: <a href="https://commits.webkit.org/297891@main">https://commits.webkit.org/297891@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c5d00d5d73d911881e51573eab44c5cba7c6dba8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113286 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33026 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23499 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119494 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63977 "Built successfully") | ✅ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33674 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41584 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86220 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ✅ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116233 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26900 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101908 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66547 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26165 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20035 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63248 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105803 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96275 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20110 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122711 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111902 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40364 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30114 "Found 1 new test failure: http/tests/site-isolation/frame-index.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95067 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40749 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98121 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94813 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39960 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17780 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36504 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18203 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40250 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45749 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136132 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39891 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36529 "Found 19 new JSC stress test failures: stress/arrowfunction-constructor.js.bytecode-cache, stress/arrowfunction-constructor.js.default, stress/arrowfunction-constructor.js.dfg-eager, stress/arrowfunction-constructor.js.dfg-eager-no-cjit-validate, stress/arrowfunction-constructor.js.eager-jettison-no-cjit, stress/arrowfunction-constructor.js.mini-mode, stress/arrowfunction-constructor.js.no-cjit-collect-continuously, stress/arrowfunction-constructor.js.no-cjit-validate-phases, stress/arrowfunction-constructor.js.no-llint, stress/builtin-function-is-construct-type-none.js.bytecode-cache ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43224 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41628 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
r+ assuming this lands after https://github.com/WebKit/WebKit/pull/48537

aprotyas committed:
Pointer Lock should compile on iOS
https://bugs.webkit.org/show_bug.cgi?id=296501
rdar://156741590

Reviewed by Richard Robinson and Aditya Keerthi.

This patch drops the build-time platform restriction, since there is
nothing in the Pointer Lock support code that prevents it from building
on most iOS family platforms.

Note that we still don't build Pointer Lock on tvOS/watchOS because of
missing platform support.

* Source/WTF/wtf/PlatformEnableCocoa.h:

Canonical link: https://commits.webkit.org/297891@main
